jupyter/nmm-minimax.ipynb

```diff
@@ -36,8 +29,21 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "`memoize(f)` takes a function and checks if the function was already called with the same two main arguments and returns the saved return value. If it is not yet in the Cache, the arguments and it's return values will be saved to the Cache\n",
+    "## Minimax\n",
     "\n",
+    "Der *Minimax* Algorithmus bietet eine einfache Möglichkeit um den perfekten Weg in einem Nullsummenspiel zu berechnen. Um diesen zu finden wird der komplette Spielbaum vollständig per Tiefensuche durchsucht.\n",
+    "Für einen Spielbaum mit der Tiefe `h` und dem Verzweigungsgrad `b` bedeutet das für die Zeit `t` und den Speicher `m`:\n",
+    "<font size=\"4\">\n",
+    "$$ t_{Minimax} \\in \\mathcal{O}(b^h) $$\n",
+    "$$ m_{Minimax} \\in \\mathcal{O}(bh) $$\n",
+    "</font>\n",
+    "Logischerweise ist *Minimax* somit nicht für die komplette Berechnung des Spiels geeignet, weil dies die üblicherweise zur Verfügung stehenden Ressourcen überschreitet. Aus diesem Grund wird die Tiefensuche auf eine maximale Tiefe `limit` beschränkt. Dies hat jedoch zur Folge, dass bei der maximalen Tiefe häufig noch kein eindeutiges Ergebnis *Mini* dem Wert `-1` (sichere Niederlage für `player`) oder *Max* mit dem Wert `1` (sicherer Sieg für `player`) erkannt werden konnte. Somit ist die Funktion `heuristic(state, player)` notwendig die in einem solchen Fall eine einfache heuristische Bewertung des `state` für den `player` durchführt und einen Wert $ -1 < value < 1 $ berechnet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
     "Die Funktion `memoize(f)` nimmt eine Funktion `f` und überprüft ob diese bereits mit den beiden gleichen Haupt-Parametern aufgerufen wurde und liefert den gespeicherten Antwortwert. Wenn die Funktion bisher nicht aufgerufen wurde, werden die Parameter und die Antwortwerte in den Cache gespeichert"
```

Comment:
Finde ich besser für den letzten Teilsatz:
"werden die Parameter und das entsprechende Ergebnis in den Cache gespeichert."